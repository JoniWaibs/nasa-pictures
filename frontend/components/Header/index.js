import React from 'react'
const { useContext } = React
import styles from '../../styles/Home.module.scss'

import context from '../../context/AppContext'
const { AppContext } = context;

import filters from '../../components/Filters'
const { CamerasFilter, DatesFilter } = filters

import { Service as ApiService } from '../../services'

const Header = () => {
  const { setIsLoading, setPictures} = useContext(AppContext);

  const getPicturesByConfig = async (customConfig) => {
    const params = customConfig
    setIsLoading(true)
    try {
      const { data } = await new ApiService(params).getPictures()
      setPictures(data)
    } catch (error) {
      setPictures([])
    } finally {
      setIsLoading(false)
    }
  }

  return (
    <header className={styles.header}>
      <h1>
        Welcome to <a href="https://nextjs.org">NASA PICS!</a>
      </h1>

      <CamerasFilter getPicturesByConfig={getPicturesByConfig}/>
      <DatesFilter getPicturesByConfig={getPicturesByConfig}/>
    </header>
  )
}

export default Header
