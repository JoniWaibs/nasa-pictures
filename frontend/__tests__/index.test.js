// __tests__/index.test.jsx
import React from 'react'
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent } from '@testing-library/react'
import Card from '../components/Card'
import Header from '../components/Header'
import filters from '../components/Filters'
const { CamerasFilter } = filters

const dogImage = "https://images.dog.ceo/breeds/puggle/IMG_081858.jpg";

describe('NASA photos App', () => {
  let propsMock;
  let getPicturesByConfig;

  beforeEach(() => {
    getPicturesByConfig = jest.fn()
    propsMock = {
      id: 102693,
      camera: {
        fullName: 'Front Hazard Avoidance Camera',
        id: 'Front Hazard Avoidance Camera',
        name: 'FHAZ',
        roverId: 5
      },
      imgSrc: 'http://mars.jpl.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/01000/opgs/edr/fcam/FLB_486265257EDR_F0481570FHAZ00323M_.JPG',
      earthDate: '2015-05-30',
      roverData: {
        id: 5,
        landingDate: '2012-08-06',
        launchDate: '2011-11-26',
        name: 'Curiosity',
        status: null
      }
    }
  })

  test('Should render Header', async () => {
    const { container, queryByText } = render(
      <Header/>
    )
    expect(container).toMatchSnapshot()
    queryByText('Welcome to NASA PICS!')
  });

  test('Should render Card', async () => {
    const { container, getByText } = render(
      <Card picture={propsMock} />
    )
    expect(container).toMatchSnapshot()
    getByText('Front Hazard Avoidance Camera')
  });

  test('Should render CamerasFilter', async () => {
    const { container, getByText } = render(
      <CamerasFilter getPicturesByConfig={getPicturesByConfig} />
    )
    expect(container).toMatchSnapshot()
    getByText('Select by Camera')

    const fhazButton = getByText('FHAZ')
    fireEvent.click(fhazButton)

    expect(getPicturesByConfig).toHaveBeenCalled()
  });
})
