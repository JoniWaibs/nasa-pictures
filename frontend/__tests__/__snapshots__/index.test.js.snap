// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NASA photos App Should render CamerasFilter 1`] = `
<div>
  <div
    class="filtersWrapper"
  >
    <p>
      Select by Camera
    </p>
    <div
      class="btnGroup"
    >
      <button
        class="selectBtn"
        type="button"
      >
        FHAZ
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        RHAZ
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        MAST
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        CHEMCAM
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        MAHLI
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        MARDI
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        NAVCAM
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        PANCAM
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        MINITES
      </button>
    </div>
  </div>
</div>
`;

exports[`NASA photos App Should render Card 1`] = `
<div>
  <div
    class="picture"
  >
    <div
      class="pictureHeader"
    >
      <p>
        Front Hazard Avoidance Camera
      </p>
    </div>
    <div
      class="imgWrapper"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg xmlns=%27http://www.w3.org/2000/svg%27 version=%271.1%27 width=%27200%27 height=%27200%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Picture of the author"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==); background-position: 0% 0%;"
        />
        <noscript />
      </span>
    </div>
  </div>
</div>
`;

exports[`NASA photos App Should render Header 1`] = `
<div>
  <header
    class="header"
  >
    <h1>
      Welcome to NASA pics!
    </h1>
    <div
      class="filtersWrapper"
    >
      <p>
        Select by Camera
      </p>
      <div
        class="btnGroup"
      >
        <button
          class="selectBtn"
          type="button"
        >
          FHAZ
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          RHAZ
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          MAST
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          CHEMCAM
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          MAHLI
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          MARDI
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          NAVCAM
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          PANCAM
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          MINITES
        </button>
      </div>
    </div>
    <div
      class="filtersWrapper"
    >
      <p>
        Select by Date
      </p>
      <div
        class="pickerContainer"
      >
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class="picker"
              type="text"
              value="02/06/2022"
            />
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`NASA photos View Should render CamerasFilter 1`] = `
<div>
  <div
    class="filtersWrapper"
  >
    <p>
      Select by Camera
    </p>
    <div
      class="btnGroup"
    >
      <button
        class="selectBtn"
        type="button"
      >
        FHAZ
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        RHAZ
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        MAST
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        CHEMCAM
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        MAHLI
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        MARDI
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        NAVCAM
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        PANCAM
      </button>
      <button
        class="selectBtn"
        type="button"
      >
        MINITES
      </button>
    </div>
  </div>
</div>
`;

exports[`NASA photos View Should render Card 1`] = `
<div>
  <div
    class="picture"
  >
    <div
      class="pictureHeader"
    >
      <p>
        Front Hazard Avoidance Camera
      </p>
    </div>
    <div
      class="imgWrapper"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg xmlns=%27http://www.w3.org/2000/svg%27 version=%271.1%27 width=%27200%27 height=%27200%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Picture of the author"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==); background-position: 0% 0%;"
        />
        <noscript />
      </span>
    </div>
  </div>
</div>
`;

exports[`NASA photos View Should render Header 1`] = `
<div>
  <header
    class="header"
  >
    <h1>
      Welcome to NASA pics!
    </h1>
    <div
      class="filtersWrapper"
    >
      <p>
        Select by Camera
      </p>
      <div
        class="btnGroup"
      >
        <button
          class="selectBtn"
          type="button"
        >
          FHAZ
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          RHAZ
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          MAST
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          CHEMCAM
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          MAHLI
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          MARDI
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          NAVCAM
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          PANCAM
        </button>
        <button
          class="selectBtn"
          type="button"
        >
          MINITES
        </button>
      </div>
    </div>
    <div
      class="filtersWrapper"
    >
      <p>
        Select by Date
      </p>
      <div
        class="pickerContainer"
      >
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class="picker"
              type="text"
              value="02/06/2022"
            />
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
